src/config/job-config/actions/defaultjobactions.module.ts

```diff
@@ -0,0 +1,90 @@
+import { Module } from "@nestjs/common";
+import { LogJobActionFactory } from "./logaction/logaction.factory";
+import { LogJobActionModule } from "./logaction/logaction.module";
+import { EmailJobActionFactory } from "./emailaction/emailaction.factory";
+import { EmailJobActionModule } from "./emailaction/emailaction.module";
+import { actionType as logActionType } from "./logaction/logaction.interface";
+import { actionType as emailActionType } from "./emailaction/emailaction.interface";
+import { ValidateJobActionModule } from "./validateaction/validateaction.module";
+import { actionType as validateActionType } from "./validateaction/validateaction.interface";
+import { ValidateJobActionFactory } from "./validateaction/validateaction.factory";
+import { URLJobActionModule } from "./urlaction/urlaction.module";
+import { URLJobActionFactory } from "./urlaction/urlaction.factory";
+import { actionType as urlActionType } from "./urlaction/urlaction.interface";
+import { RabbitMQJobActionModule } from "./rabbitmqaction/rabbitmqaction.module";
+import { actionType as rabbitmqActionType } from "./rabbitmqaction/rabbitmqaction.interface";
+import { RabbitMQJobActionFactory } from "./rabbitmqaction/rabbitmqaction.factory";
+import {
+  CREATE_JOB_ACTION_FACTORIES,
+  UPDATE_JOB_ACTION_FACTORIES,
+} from "../jobconfig.interface";
+
+/**
+ * Provide a list of built-in job action factories.
+ *
+ * CREATE_JOB_ACTION_FACTORIES and UPDATE_JOB_ACTION_FACTORIES be extended (eg by a
+ * plugin) with additional factories.
+ */
+@Module({
+  imports: [
+    EmailJobActionModule,
+    LogJobActionModule,
+    ValidateJobActionModule,
+    URLJobActionModule,
+    RabbitMQJobActionModule,
+  ],
+  providers: [
+    {
+      provide: CREATE_JOB_ACTION_FACTORIES,
+      useFactory: (
+        logJobActionFactory,
+        emailJobActionFactory,
+        validateJobActionFactory,
+        urlJobActionFactory,
+        rabbitMQJobActionFactory,
```

Comment:
Maybe I am misunderstanding this or the name `defaultjobactions` is confusing me, but by doing this don't we by default enable all actions that are written here? Instead we should only enable those that are declared in the jobConfig file. Perhaps if this was split to `default` and `additional` actions it would be clearer? Where in `additional` someone could declare whether the use rabbitMQ or Kafka for example, instead of by default having rabbitMQ

Reply:
When the yaml file gets parsed you need to have a mapping between `actionType` strings and the `JobActionFactory` instance that will parse that part of the config and create a `JobAction`. Would it be clearer if the file were called `defaultjobactionfactories.module.ts` like the module? Somehow the name should reflect that this is the stuff that gets created before the yaml file is parsed, not the stuff configured by the yaml.

If you wanted to add new actions (say using a plugin) you would provide new lists for `CREATE_JOB_ACTION_FACTORIES` and `UPDATE_JOB_ACTION_FACTORIES`. I guess this could be split into the built-in ones and `additional` factories loaded from plugins, but it's not clear to me what you would gain from that.

Reply:
I've renamed this to `JobActionCreator`.